@stephentoub @ViktorHofer do you know why the analyzer kicked for `IDE0200` on a file that wasn't changed? It says:

> ##[error]src\libraries\System.Private.CoreLib\src\System\Diagnostics\Tracing\RuntimeEventSource.cs(91,75): error IDE0200: (NETCORE_ENGINEERING_TELEMETRY=Build) Lambda expression can be removed

In here 

https://github.com/dotnet/runtime/blob/7d148d2f350dea8580aad34527ec97df642f9ba8/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSource.cs#L78

I see how that lambda (and a few others there) is no longer needed, but I don't see why it wasn't flagged before. I also can't reproduce the warnings when running locally with the same command as one of the failing legs.

--- a/src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetCpuUtilization.cs
+++ b/src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetCpuUtilization.cs
@@ -16,6 +16,6 @@ internal static partial class Interop
         }
 
         [LibraryImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetCpuUtilization")]
-        internal static partial int GetCpuUtilization(ref ProcessCpuInformation previousCpuInfo);
+        internal static partial double GetCpuUtilization(ref ProcessCpuInformation previousCpuInfo);
     }
 }

--- a/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSource.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSource.cs
@@ -88,7 +88,7 @@ namespace System.Diagnostics.Tracing
                 // overhead by at all times even when counters aren't enabled.
 
                 // On disable, PollingCounters will stop polling for values so it should be fine to leave them around.
-                _cpuTimeCounter ??= new PollingCounter("cpu-usage", this, () => RuntimeEventSourceHelper.GetCpuUsage()) { DisplayName = "CPU Usage", DisplayUnits = "%" };
+                _cpuTimeCounter ??= new PollingCounter("cpu-usage", this, RuntimeEventSourceHelper.GetCpuUsage) { DisplayName = "CPU Usage", DisplayUnits = "%" };
                 _workingSetCounter ??= new PollingCounter("working-set", this, () => ((double)Environment.WorkingSet / 1_000_000)) { DisplayName = "Working Set", DisplayUnits = "MB" };
                 _gcHeapSizeCounter ??= new PollingCounter("gc-heap-size", this, () => ((double)GC.GetTotalMemory(false) / 1_000_000)) { DisplayName = "GC Heap Size", DisplayUnits = "MB" };
                 _gen0GCCounter ??= new IncrementingPollingCounter("gen-0-gc-count", this, () => GC.CollectionCount(0)) { DisplayName = "Gen 0 GC Count", DisplayRateTimeScale = new TimeSpan(0, 1, 0) };

--- a/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSourceHelper.Unix.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSourceHelper.Unix.cs
@@ -7,7 +7,7 @@ namespace System.Diagnostics.Tracing
     {
         private static Interop.Sys.ProcessCpuInformation s_cpuInfo;
 
-        internal static int GetCpuUsage() =>
+        internal static double GetCpuUsage() =>
             Interop.Sys.GetCpuUtilization(ref s_cpuInfo) / Environment.ProcessorCount;
     }
 }

--- a/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSourceHelper.Windows.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSourceHelper.Windows.cs
@@ -10,11 +10,11 @@ namespace System.Diagnostics.Tracing
         private static long s_prevSystemUserTime;
         private static long s_prevSystemKernelTime;
 
-        internal static int GetCpuUsage()
+        internal static double GetCpuUsage()
         {
             // Returns the current process' CPU usage as a percentage
 
-            int cpuUsage = 0;
+            double cpuUsage = 0.0;
 
             if (Interop.Kernel32.GetProcessTimes(Interop.Kernel32.GetCurrentProcess(), out _, out _, out long procKernelTime, out long procUserTime) &&
                 Interop.Kernel32.GetSystemTimes(out _, out long systemUserTime, out long systemKernelTime))
@@ -25,7 +25,7 @@ namespace System.Diagnostics.Tracing
                 if (s_prevSystemUserTime != 0 && s_prevSystemKernelTime != 0 && // These may be 0 when we report CPU usage for the first time, in which case we should just return 0.
                     totalSystemTime != 0)
                 {
-                    cpuUsage = (int)(totalProcTime * 100 / totalSystemTime);
+                    cpuUsage = (totalProcTime * 100.0 / totalSystemTime);
                 }
 
                 s_prevProcUserTime = procUserTime;

--- a/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.CpuUtilizationReader.Unix.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.CpuUtilizationReader.Unix.cs
@@ -9,7 +9,7 @@ namespace System.Threading
         {
             private Interop.Sys.ProcessCpuInformation _cpuInfo;
 
-            public int CurrentUtilization =>
+            public double CurrentUtilization =>
                 Interop.Sys.GetCpuUtilization(ref _cpuInfo) / Environment.ProcessorCount;
         }
     }

--- a/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.GateThread.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.GateThread.cs
@@ -102,7 +102,7 @@ namespace System.Threading
                                 (uint)threadPoolInstance.GetAndResetHighWatermarkCountOfThreadsProcessingUserCallbacks());
                         }
 
-                        int cpuUtilization = cpuUtilizationReader.CurrentUtilization;
+                        int cpuUtilization = (int)cpuUtilizationReader.CurrentUtilization;
                         threadPoolInstance._cpuUtilization = cpuUtilization;
 
                         bool needGateThreadForRuntime = ThreadPool.PerformRuntimeSpecificGateActivities(cpuUtilization);

--- a/src/native/libs/System.Native/pal_time.c
+++ b/src/native/libs/System.Native/pal_time.c
@@ -137,7 +137,7 @@ double SystemNative_GetCpuUtilization(ProcessCpuInformation* previousCpuInfo)
     double cpuUtilization = 0.0;
     if (cpuTotalTime > 0 && cpuBusyTime > 0)
     {
-        cpuUtilization = (cpuBusyTime * 100.0 / cpuTotalTime);
+        cpuUtilization = ((double)cpuBusyTime * 100.0 / (double)cpuTotalTime);
     }
 
     previousCpuInfo->lastRecordedCurrentTime = currentTime;

--- a/src/native/libs/System.Native/pal_time.c
+++ b/src/native/libs/System.Native/pal_time.c
@@ -95,7 +95,7 @@ uint64_t SystemNative_GetTimestamp()
 #endif
 }
 
-int32_t SystemNative_GetCpuUtilization(ProcessCpuInformation* previousCpuInfo)
+double SystemNative_GetCpuUtilization(ProcessCpuInformation* previousCpuInfo)
 {
     uint64_t kernelTime = 0;
     uint64_t userTime = 0;
@@ -109,7 +109,7 @@ int32_t SystemNative_GetCpuUtilization(ProcessCpuInformation* previousCpuInfo)
     else
     {
         kernelTime =
-            ((uint64_t)(resUsage.ru_stime.tv_sec) * SecondsToNanoSeconds) + 
+            ((uint64_t)(resUsage.ru_stime.tv_sec) * SecondsToNanoSeconds) +
             ((uint64_t)(resUsage.ru_stime.tv_usec) * MicroSecondsToNanoSeconds);
         userTime =
             ((uint64_t)(resUsage.ru_utime.tv_sec) * SecondsToNanoSeconds) +
@@ -134,10 +134,10 @@ int32_t SystemNative_GetCpuUtilization(ProcessCpuInformation* previousCpuInfo)
         cpuBusyTime = (userTime - lastRecordedUserTime) + (kernelTime - lastRecordedKernelTime);
     }
 
-    int32_t cpuUtilization = 0;
+    double cpuUtilization = 0.0;
     if (cpuTotalTime > 0 && cpuBusyTime > 0)
     {
-        cpuUtilization = (int32_t)(cpuBusyTime * 100 / cpuTotalTime);
+        cpuUtilization = (cpuBusyTime * 100.0 / cpuTotalTime);
     }
 
     previousCpuInfo->lastRecordedCurrentTime = currentTime;

--- a/src/native/libs/System.Native/pal_time.h
+++ b/src/native/libs/System.Native/pal_time.h
@@ -48,4 +48,4 @@ PALEXPORT uint64_t SystemNative_GetTimestamp(void);
  * returned is sum of utilization across all processors, e.g. this function will
  * return 200 when two cores are running at 100%.
  */
-PALEXPORT int32_t SystemNative_GetCpuUtilization(ProcessCpuInformation* previousCpuInfo);
+PALEXPORT double SystemNative_GetCpuUtilization(ProcessCpuInformation* previousCpuInfo);

